api.ts

```diff
@@ -334,4 +334,26 @@ namespace tileUtil {
             for (const cb of createdHandlers) cb.handler(sprite)
         }
     }
+
+    /**
+     * Loops over each tile in a tilemap and runs the nested code
+     *
+     * @param tilemap The tilemap to loop over
+     * @param handler The code to run
+     */
+    //% blockId=tileUtil_forEachTileInMap
+    //% block="for each tile in $tilemap with $column $row $location"
+    //% tilemap.shadow=tileUtil_getLoadedMap
+    //% handlerStatement
+    //% draggableParameters="reporter"
+    //% group=Tiles
+    //% weight=0
+    //% help=github:arcade-tile-util/docs/for-each-tile-in-map
+    export function forEachTileInMap(tilemap: tiles.TileMapData, handler: (column: number, row: number, location: tiles.Location) => void) {
```

Comment:
Could we also make "tile" be a draggable param like column, row, and location? Seems like it might be useful, but nbd if it's more trouble than its worth, or if you don't think it's actually all that useful.

Reply:
it's usually easier to just use the "tile at location is image" block from the scene category. there isn't much you can do with the image itself since you have to drag out an image equals block anyways (and that's under advanced)